tests/coverage/test_coverage.py

```diff
@@ -0,0 +1,92 @@
+# SPDX-License-Identifier: Apache-2.0
+from antlr4 import InputStream
+from tests.coverage.case_file_parser import parse_stream, parse_one_file
+from tests.coverage.nodes import CaseLiteral
+
+
+def parse_string(input_string):
+    return parse_stream(InputStream(input_string))
+
+
+def test_parse_basic_example():
+    test_file = parse_string(
+        """### SUBSTRAIT_SCALAR_TEST: v1.0
```

Comment:
let's create a util method to build the header so the test cases are just the specific stuff that we care about in that case.